### PR TITLE
Add initial config for CLA Assistant GH Action

### PR DESCRIPTION
**Pull request by mads-hartmann**
## Description

This automates our CLA process through the use of [cla-assistant/github-action](https://github.com/cla-assistant/github-action). The signatures are stored in a private repository (gitpod-io/cla).

The message used to prompt the signing of the CLA has been modified so that it makes a distinction between individual users and companies. We are only automating the CLA flow for individuals now as the process for signing on behalf of a company is more complicated and it doesn't happen very often.

TODO
- [x] Update `path-to-document` so it points to our CLA  
  The CLA is stored in gitpod-io/cla and exposed using GitHub Pages
- [ ] Update the CLA to reflect new process (see [PR](https://github.com/gitpod-io/cla/pull/1))
- [x]  Create an access token and add it to secrets using the name `CLA_ACTION_ACCESS_TOKEN` (see [docs](https://github.com/cla-assistant/github-action#6-adding-personal-access-token-as-a-secret))
- [x] Add any bots that we use to the `allowlist` (see [docs](https://github.com/cla-assistant/github-action#5-users-and-bots-in-allowlist))
- [x] Either add all Gitpod employees to the `allowlist` or ~decide that we'll just have everyone sign the CLA on the next PR~
- [x] Either add all users who have already signed the CLA to the `allowlist` ~or decide that we'll just have everyone sign the CLA again~
- [ ] Verify that having people "sign" the CLA by leaving a GitHub Pull Request Comment is sufficient legally for us (@meysholdt is verifying this)

## Related Issue(s)

Fixes #gitpod-io/ops#130

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mrsimonemms** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mrsimonemms` in a comment when ready.

Associated issue: *#6*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: .github/workflows/cla.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          # Where to store the signatures
+          remote-organization-name: gitpod-io
+          remote-repository-name: cla
+          branch: "main"
+          path-to-signatures: "signatures/v1/cla.json"
+          # messages for the bot to post
+          custom-notsigned-prcomment: |
+            Thank you for your submission, we really appreciate it.
+
+            Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://gitpod-io.github.io/cla/CLA.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.
```
Comment (ghuntley) [inline .github/workflows/cla.yml:28]:
```suggestion
            Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://www.gitpod.io/cla) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.
```

File: .github/workflows/cla.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          custom-pr-sign-comment: "I have read the CLA Document and I hereby sign the CLA"
+          # The following GitHub users don't have to sign the CLA
+          # - Gitpod employees (https://github.com/orgs/gitpod-io/teams/team-gitpod/members)
+          # - Bots
+          # - Anyone who signed the CLA manually prior to the use of CLA Assistant
+          #   Filip Troníček (@filiptronicek)
+          #   Jordan Hailey (@jordanhailey)
+          #   Ofer Shaal (@shaal),
+          #   Pudong Zheng (@iQQBot)
+          allowlist: >
```
Comment (ghuntley) [inline .github/workflows/cla.yml:40]: thoughts: drop all people from this list except robots. Use this list only for robots which cannot do the /sign process. Asking folks to resign isn't a concern is it?

Comment (mads-hartmann) [reply .github/workflows/cla.yml:40]: @ghuntley I'll remove all the existing signers (it is probably out of date already anyway). For the bots  I'll keep them for the first iteration so we can get this out the door and then we can simplify it later.

**Comment by mads-hartmann** [comment]
@ghuntley Thanks for the fixes and comment!

**Comment by tnir** [comment]
@mads-hartmann Any updates on this after 6 month from inception?

**Comment by tnir** [comment]
@mads-hartmann Any updates on this after 9 month from inception?

**Comment by tnir** [comment]
Or @meysholdt?

**Comment by meysholdt** [comment]
I'm planning to pick this up again this week.

**Comment by meysholdt** [comment]
Closing this PR as we now have a solution based on DocuSign PowerForms.

**Pull request closed without merging**